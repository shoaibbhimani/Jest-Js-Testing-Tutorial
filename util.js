var addItemToList = function(list,newItem){
 return list.concat([newItem]);
}

var removeItemFromList = function(list,index){
  return list.filter(function(listItem,i){
  	return i !== index
  })
}

var updateItemFromList = function(list,index,newItem){
  // Clone Array 
  var newList = list.slice();
  //new Value add 
  newList[index] = newItem
  //new array return 
  return newList;
  
}

var addNewPropertyToObject = function(person,newItem){
 // person['name'] = newItem;
 // return Object.assign({}, person, { name:newItem })
 return {...person, name:newItem}
}

module.exports = {
	addItemToList,
	removeItemFromList,
	updateItemFromList,
	addNewPropertyToObject
}