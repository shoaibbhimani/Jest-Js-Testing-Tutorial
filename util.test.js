import util from './util.js';
// import example from './example.js';

import deepFreeze from "deep-freeze"

describe("Array CRUD Test Cases",() => {
	test("Add Item to a List", () => {
		var jslibs = ['React', 'Mobx'];
		deepFreeze(jslibs);
		var result = util.addItemToList(jslibs, 'Redux');
		// var expected = ['React', 'Mobx', 'Jest']
		expect(result).toMatchSnapshot();
	})

	test("Remove Item from a List", () => {
		var jslibs = ['React', 'Mobx'];
		deepFreeze(jslibs);
		var result = util.removeItemFromList(jslibs, 1);
		// var expected = ['React']
		expect(result).toMatchSnapshot()
	})

	test("Update Item from a List", () => {
		var jslibs = ['React', 'Mobx'];
		deepFreeze(jslibs);
		var result = util.updateItemFromList(jslibs, 1, 'Redux');
		// var expected = ['React','Redux']
		expect(result).toMatchSnapshot()
	})
})


describe("Objects Test Cases", () => {
	test("Add property to object", () => {
		var person = { age:25  }
		deepFreeze(person);
		var result = util.addNewPropertyToObject(person, 'Shoaib');
		var expected = { name:"Shoaib", age:25 }
		expect(result).toEqual(expected)
	})
})