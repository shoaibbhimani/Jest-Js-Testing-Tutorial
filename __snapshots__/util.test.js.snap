// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array CRUD Test Cases Add Item to a List 1`] = `
Array [
  "React",
  "Mobx",
  "Redux",
]
`;

exports[`Array CRUD Test Cases Remove Item from a List 1`] = `
Array [
  "React",
]
`;

exports[`Array CRUD Test Cases Update Item from a List 1`] = `
Array [
  "Redux",
  "Mobx",
]
`;
